reader.js

```diff
@@ -7,15 +9,156 @@ const CLOSE_TIMEOUT = 10000;
 
 let WSS;
 
-function init(wss) {
+function init(server) {
     console.log('Initializing nasonov-reader');
+    const wss = new WebSocket.Server({server});
 
     WSS = wss;
     wss.on('connection', function connected(ws, req) {});
 
+    server.on('request', (request, response) => {connectToInflux(request, response)});
+
     connectToWriter();
 }
 
+function connectToInflux(request, response) {
+
+    if(request.method === 'GET') {
+
+        let requestQuery = new URL(request.url, 'https://habmc.stanfordssi.org/');
+
+        if(requestQuery.pathname.search('/index') > 0) {
+
+            let mission = requestQuery.pathname.substring(1, requestQuery.pathname.search('/index'));
+
+            if (isNaN(mission)) {
+                return;
+            }
+
+            influxConnection.then(function (influx) {
+
+                influx.getMeasurements().then(names => {
+
+                    let query = `select * from`;
+                    names.forEach(name => {
+                        query += ` ${name},`;
+                    });
+                    query = query.slice(0, -1);
+                    query += ` where mission = '${mission}'`;
+
+                    influx.query(query).then(result => {
+
+                        let ids = {};
+                        result.forEach(measure => {
+                            if (!ids.hasOwnProperty(measure.id)) {
+                                ids[measure.id] = measure.time;
+                            }
+                        });
+
+                        response.setHeader("Access-Control-Allow-Origin", '*');
+                        response.end(JSON.stringify(ids));
+
+                    }).catch(function (err) {
+                        console.error(`Error querying data from InfluxDB! ${err.stack}`);
+
+                        if (WSS.readyState !== WebSocket.OPEN) {
+                            return;
+                        }
+
+                        WSS.send(id + ':error:' + e);
```

Comment:
We should send the error over HTTP with a 500 status code, rather than than over the websocket.